llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -2466,7 +2496,9 @@ class BoUpSLP {
         }
         // TODO: Check if we can remove a check for non-power-2 number of
         // scalars after full support of non-power-2 vectorization.
-        return UniqueValues.size() != 2 && has_single_bit(UniqueValues.size());
+        return UniqueValues.size() != 2 &&
+               hasFullVectorsOnly(*R.TTI, (*UniqueValues.begin())->getType(),
```

Comment:
Instead of white listing only the new class here, can we remove the restriction in a separate patch?  This should be fully VL independent right?

Reply:
Currently no, it will regress some code. I 'll restore an original check here and will update it later